src/components/dialog/dialog.pw.tsx

```diff
@@ -0,0 +1,596 @@
+
+    await thirdTextbox.press("Shift+Tab");
+    await expect(secondTextbox).toBeFocused();
+
+    await secondTextbox.press("Shift+Tab");
+    await expect(firstTextbox).toBeFocused();
+
+    await firstTextbox.press("Shift+Tab");
+    await expect(closeButton).toBeFocused();
+
+    await closeButton.press("Shift+Tab");
+    await expect(thirdTextbox).toBeFocused();
+  });
+
+  test("when tabbing through Dialog content, background should not scroll to the bottom of the page", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogBackgroundScrollTest />);
+    const dialog = page.getByRole("dialog");
+    const textbox = page.getByLabel("Textbox");
+    const closeButton = page.getByLabel("Close");
+
+    await dialog.press("Tab");
+    await closeButton.press("Tab");
+    await textbox.press("Tab");
+
+    await expect(closeButton).toBeFocused();
+    await expect(
+      page.getByText("I should not be scrolled into view")
+    ).not.toBeInViewport();
+  });
+
+  test("when shift tabbing through Dialog content, background should not scroll to the bottom of the page", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogBackgroundScrollTest />);
+    const dialog = page.getByRole("dialog");
+    const textbox = page.getByLabel("Textbox");
+    const closeButton = page.getByLabel("Close");
+
+    await dialog.press("Shift+Tab");
+    await textbox.press("Shift+Tab");
+
+    await expect(closeButton).toBeFocused();
+    await expect(
+      page.getByText("I should not be scrolled into view")
+    ).not.toBeInViewport();
+  });
+
+  test.describe(
+    "when there is a button inside Dialog, which opens a Toast",
+    () => {
+      test("clicking button moves focus out of Dialog to the newly-opened Toast", async ({
+        mount,
+        page,
+      }) => {
+        await mount(<DialogWithToast />);
+        const openToastButton = page
+          .getByRole("button")
+          .filter({ hasText: "Open Toast" });
+        await openToastButton.click();
+
+        const toast = toastComponent(page);
+        await expect(toast).toBeFocused();
+      });
+
+      test("when Toast is opened and focus on it is lost, pressing Tab key traps focus back inside the Dialog", async ({
+        mount,
+        page,
+      }) => {
+        await mount(<DialogWithToast />);
+        const dialog = page.getByRole("dialog");
+        const openToastButton = page
+          .getByRole("button")
+          .filter({ hasText: "Open Toast" });
+        await openToastButton.click();
+
+        await page.mouse.click(0, 0); // click outside Toast and Dialog
+        await dialog.press("Tab");
+
+        await expect(openToastButton).toBeFocused();
+      });
+
+      test("when tabbing through Dialog content and two opened Toasts, the background scroll should not scroll to the bottom of the page", async ({
+        mount,
+        page,
+      }) => {
+        await mount(<DialogWithOpenToastsBackgroundScrollTest />);
```

Comment:
suggestion: (non-blocking) it would be better to add a space to separate the mount from the assertion to follow the same pattern as with the other test cases.